test/integration/sagemaker/test_mnist.py

```diff
@@ -19,24 +19,38 @@
 import sagemaker
 from sagemaker.pytorch import PyTorchModel
 
-from test.integration import mnist_script, model_cpu_dir
+from test.integration import model_cpu_dir, mnist_cpu_script, mnist_gpu_script, model_eia_dir, mnist_eia_script
 from test.integration.sagemaker.timeout import timeout_and_delete_endpoint
 
 
 @pytest.mark.skip_gpu
+@pytest.mark.skip_eia
 def test_mnist_distributed_cpu(sagemaker_session, ecr_image, instance_type):
     instance_type = instance_type or 'ml.c4.xlarge'
-    _test_mnist_distributed(sagemaker_session, ecr_image, instance_type)
+    _test_mnist_distributed(sagemaker_session, ecr_image, instance_type, 'cpu', mnist_cpu_script)
 
 
 @pytest.mark.skip_cpu
+@pytest.mark.skip_eia
 def test_mnist_distributed_gpu(sagemaker_session, ecr_image, instance_type):
     instance_type = instance_type or 'ml.p2.xlarge'
-    _test_mnist_distributed(sagemaker_session, ecr_image, instance_type)
+    _test_mnist_distributed(sagemaker_session, ecr_image, instance_type, 'gpu', mnist_gpu_script)
 
 
-def _test_mnist_distributed(sagemaker_session, ecr_image, instance_type):
-    model_dir = os.path.join(model_cpu_dir, 'model_mnist.tar.gz')
+@pytest.mark.skip_cpu
+@pytest.mark.skip_gpu
+def test_mnist_eia(sagemaker_session, ecr_image, instance_type):
+    instance_type = instance_type or 'ml.c4.xlarge'
+    _test_mnist_distributed(sagemaker_session, ecr_image, instance_type, 'eia', mnist_eia_script)
+
+
+def _test_mnist_distributed(sagemaker_session, ecr_image, instance_type, processor, mnist_script):
+    if processor == 'cpu' or processor == 'gpu':
+        model_dir = os.path.join(model_cpu_dir, 'model_mnist.tar.gz')
+    elif processor == 'eia':
```

Comment:
Based on my understanding, using EIA within the SageMaker platform requires supplying a parameter for the `accelerator_type` argument to the `deploy` function, which I believe isn't present in this PR or the current existing code right now.

Reply:
Thanks for pointing this out! I've updated the `deploy()` call to have an `accelerator_type` argument. Spoke with Asimov team and they are in favor of keeping the test logic for EI bundled with the current CPU and GPU files